### PR TITLE
added in a 404 page for routes not found

### PR DESCRIPTION
**Pull request by niccheww**
## Hey, I just made a Pull Request!

This is a simple plugin to show the 404 page when the route is not found.
Screenshot 2020-09-18 at 10 14 50 AM

Please contact the Grab's engineering team for more details.
@althafh @lowjoel @feiming 

Please note:
Due to _react-router_ configurations, where the \* always matches, please ensure that this plugin is the **last** registered route

#### :heavy_check_mark: Checklist

- [ x] All tests are passing `yarn test`
- [ x] Screenshots attached (for UI changes)
- [ ] Relevant documentation updated
- [ x] Prettier run on changed files
- [ x] Tests added for new functionality
- [ ] Regression tests added for bug fixes

### CODE REVIEW

**Comment by althafh** [comment]
The 404 Image is an internal one used by Grab. Happy to get it changed to a more generic one by the Backstage Design team.

File: plugins/not-found/package.json [hunk 1]
```diff
@@ -0,0 +1,46 @@
+{
+  "name": "@backstage/plugin-not-found",
+  "version": "0.1.1-alpha.20",
```
Comment (lowjoel) [inline plugins/not-found/package.json:3]: I think the version numbers should all be set to the same as every other plugin.

Comment (niccheww) [reply plugins/not-found/package.json]: Thanks, updated the version to `0.1.1-alpha.21`

**Review by stefanalund — CHANGES_REQUESTED** [review]
Thanks! But this is not the artwork that we should use.

@katz95 can you point to the existing "not found" illustrations that we have?

**Comment by Fox32** [comment]
Should this really be a plugin, or isn't that a feature for core?

**Comment by freben** [comment]
I would argue the same - could it make sense to make a `Default404Page` component in `packages/core`, and add the route oneliner in both the example-app and the generated app skeleton?

**Comment by freben** [comment]
At the bottom here https://github.com/spotify/backstage/blob/master/packages/app/src/App.tsx#L66

and here https://github.com/spotify/backstage/blob/master/packages/create-app/templates/default-app/packages/app/src/App.tsx#L42

**Comment by Rugvip** [comment]
The 404 error page is already a part of the core App API, see the components that you can register here: https://github.com/spotify/backstage/blob/f7a57e89d0d3ee9fd1e80839d5a65fc65de974b9/packages/core-api/src/app/types.ts#L55

The default component is here: https://github.com/spotify/backstage/blob/master/packages/core/src/layout/ErrorPage/ErrorPage.tsx, which is wired together here: https://github.com/spotify/backstage/blob/f7a57e89d0d3ee9fd1e80839d5a65fc65de974b9/packages/core/src/api-wrappers/createApp.tsx#L101

You can provide your own 404 page with the following:

```tsx
const app = createApp({
  apis,
  plugins: Object.values(plugins),
  components: {
    NotFoundErrorPage: CustomNotFoundPage,
  },
});
```

Now that entire process does need more documentation, so would gladly accept contributions around that! :grin:

**Comment by Rugvip** [comment]
Should also explain why it's part of the core API: The components registered as a part of `createApp` are intended to be used as globally replaceable components, so that an app can replace them wherever they are used, be it in internal, or other open source plugins.

If the 404 page is a plain component, or a plugin that's used by other plugins, then there would be no way for an app to override the page that is used, and you would have to start making it configurable for each plugin. For example, if the `catalog` plugin used this this plugin as it's 404 page, how would you override it with your own design in your own app?

**Comment by niccheww** [comment]
Thanks Rugvip, Freben for the advise regarding the 404 page. Please advise me as to how I may be of assistance in the documentation else I will be closing the PR as it's not necessary. Thanks.

**Pull request closed without merging**